Test failures reported upstream at https://github.com/ron-rs/ron/issues/592

Compared to versoin 0.7.1, which is the version in testin at the time of
writing, these tests seem to be newly added tests rather than regressions.

Index: rust-ron-0.12.0/src/de/tests.rs
===================================================================
--- rust-ron-0.12.0.orig/src/de/tests.rs
+++ rust-ron-0.12.0/src/de/tests.rs
@@ -672,6 +672,7 @@ fn test_any_number_precision() {
     check_de_any_number("+.3", 0.3_f64);
     check_de_any_number("0.3", 0.3_f64);
     check_de_any_number("NaN", f32::NAN);
+    #[cfg(not(target_arch="riscv64"))]
     check_de_any_number("-NaN", -f32::NAN);
     check_de_any_number("inf", f32::INFINITY);
     check_de_any_number("-inf", f32::NEG_INFINITY);
@@ -708,6 +709,7 @@ fn test_value_special_floats() {
         from_str("+NaN"),
         Ok(Value::Number(Number::F32(f32::NAN.into())))
     );
+    #[cfg(not(target_arch="riscv64"))]
     assert_eq!(
         from_str("-NaN"),
         Ok(Value::Number(Number::F32((-f32::NAN).into())))
